partcad/src/partcad/assembly_factory_assy.py

```
@@ -845,57 +675,31 @@
-                        )
-                    if (
-                        connect_with_port is not None
-                        and connect_with_port_pattern is not None
-                    ):
-                        if not fnmatch.fnmatch(
-                            connect_with_port, connect_with_port_pattern
-                        ):
+                        source_port = item.with_ports.get_ports()[connect_with_port]
+                        pc_logging.debug("Found source port: %s" % source_port.name)
+                    if connect_with_port is not None and connect_with_port_pattern is not None:
+                        if not fnmatch.fnmatch(connect_with_port, connect_with_port_pattern):
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)